src/bin/commands/shard.rs

```diff
@@ -0,0 +1,368 @@
+use crate::commands::command::Command;
+use anyhow::{Result, anyhow};
+use clap::Parser;
+use fgoxide::io::Io;
+use fgoxide::iter::IntoChunkedReadAheadIterator;
+use itertools::Itertools;
+use log::info;
+use pooled_writer::{Pool, PoolBuilder, PooledWriter, bgzf::BgzfCompressor};
+use proglog::{CountFormatterKind, ProgLogBuilder};
+use seq_io::fastq::OwnedRecord;
+use seq_io::fastq::Reader as FastqReader;
+use seq_io::fastq::Record;
+use std::fs::File;
+use std::io::{BufRead, BufWriter, Write};
+use std::{
+    io,
+    path::{Path, PathBuf},
+};
+
+/// Buffer size used when opening BufReads for input files
+const BUFFER_SIZE: usize = 64 * 1024;
+
+/// Struct to hold all the output writers for a single shard
+struct ShardWriters<W: Write> {
+    /// Name of the sample this set of writers is for
```

Comment:
Stale doc? It is more like an ID than it is a name.


```suggestion
    /// ID of the shard this set of writers is for
```